api/ccapi/include/layer.h

```diff
@@ -133,7 +133,7 @@ class Layer {
    * @note      This name might be changed once this layer is added to the model
    * to keep the name unique to the model
    */
-  virtual std::string getName() noexcept = 0;
+  virtual std::string getName() const noexcept = 0;
```

Comment:
as long we are updating, how about 
```suggestion
  virtual const std::string &getName() const noexcept = 0;
```

Reply:
just const is fine.

Reply:
Done